internal/core/layout.rs

```diff
@@ -625,27 +625,14 @@ pub fn solve_path_layout(
     repeater_indexes: Slice<u32>,
 ) -> SharedVector<Coord> {
     use lyon_geom::*;
-    use lyon_path::iterator::PathIterator;
+    use lyon_path::PathEvent;
 
     // Clone of path elements is cheap because it's a clone of underlying SharedVector
     let mut path_iter = data.elements.clone().iter();
     path_iter.fit(data.width as _, data.height as _, None);
 
     let tolerance: f32 = 0.1; // lyon::tessellation::StrokeOptions::DEFAULT_TOLERANCE
-
-    let segment_lengths: Vec<_> = path_iter
-        .iter()
-        .bezier_segments()
-        .map(|segment| match segment {
-            BezierSegment::Linear(line_segment) => line_segment.length(),
-            BezierSegment::Quadratic(quadratic_segment) => {
-                quadratic_segment.approximate_length(tolerance)
-            }
-            BezierSegment::Cubic(cubic_segment) => cubic_segment.approximate_length(tolerance),
-        })
-        .collect();
-
-    let path_length: f32 = segment_lengths.iter().sum();
+    let path_length = lyon_algorithms::length::approximate_length(path_iter.iter(), tolerance);
```

Comment:
Nice :)

Reply:
Before, this was computing the length and adding the stuff to the vector. Now this is done in two steps